### ISSUE TITLE
🐛 Bug Report: Search queries do not respect quotes

### ISSUE THREAD

**Issue by starryeyez024** [backstage/backstage#26392]
### 📜 Description

When executing a search in Backstage, it should respect quote marks which encapsulate a string, and only return results that match that full string. For example, if you search for "row your boat" it should only return entities with "row your boat" in the name or description. 
Also the same results are returned in the quick results modal as are shown on the full results page.

### 👍 Expected behavior

If you search for "digital acceleration team" it should only return entities with the full string "digital acceleration team" in the name or description.

### 👎 Actual Behavior with Screenshots

If you search for "digital acceleration team", it will return entities which only have "accelerate" in the name.
Adding quote marks around a sequence of words makes no difference.

<img width="1211" alt="no_quotes" src="https://github.com/user-attachments/assets/4d77e34f-8490-48b2-ae38-49ba14ecd0dc">
<img width="1245" alt="quotes" src="https://github.com/user-attachments/assets/0dbaef06-c201-400b-b3de-544aaad538f0">

### 👟 Reproduction steps

Try searching for a string with and without quotes and notice there is no distinction. 

### 🖥️ Your Environment

 Backstage | 1.28.2  

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by benjdlambert** [comment]
Ping @backstage/search-maintainers

**Comment by githoober** [comment]
@starryeyez024  Did you end up somehow pinging @backstage/search-maintainers? I could not find how to do it myself.

**Comment by awanlin** [comment]
Hi @starryeyez024, Ben was doing just that in his comment and you did just now. The team you just mentioned would get an email. As they are pretty small team they may not have been able to get to this. Let me see what I can do to get their input on this.

**Comment by starryeyez024** [comment]
Thanks, I didn't realize I should do that.

**Comment by awanlin** [comment]
No worries, I don't think it was expected of you @starryeyez024 👍 

While I have you what "search engine" are you using Postgres or Elasticsearch? I'm pretty sure supporting this for Postgres will require some changes specific to it's implementation.

**Comment by starryeyez024** [comment]
We are using Postgresql.

**Comment by githoober** [comment]
The same is true when using Elasticsearch.

![image](https://github.com/user-attachments/assets/8c3833f0-93b3-40a0-bfd6-dd245b1ccdb7)

Honestly, this makes the search quite useless. 

Maybe, there is a workaround before it gets fixed?

**Comment by squid-ney** [comment]
Hi all, thanks for reporting this. Search should respect the quotation marks in the search term. Thanks @githoober for narrowing down the issue. The @backstage/search-maintainers will take a look and propose a PR.

**Comment by awanlin** [comment]
Pretty sure we would need to make some changes to the Postgres side as well. I think we want to leverage `websearch_to_tsquery` but I'm not sure exactly how to make that work with what we have in place today, docs on this are here: https://www.postgresql.org/docs/current/textsearch-controls.html

**Comment by squid-ney** [comment]
I'm working on a fix for the issue in Elasticsearch! PR: #27720
  
Regarding this bug for Postgres, I'm unable to reproduce. Are you doing any customization? Internally, we don't use the `PgSearchEngine` so I'm not as familiar. However, when I set it up locally, I noticed every search is treated as a phrase query by default. All search terms are combined with the "AND" operator. Here are a few examples of what I tested locally:

1. "hello world" search
<img width="1241" alt="Screenshot 2024-11-22 at 7 47 43 AM" src="https://github.com/user-attachments/assets/1b05aee0-7e14-4d40-8d4c-5b13925c9810">
PgTerm from log: ("hello"` | "hello":*)&("world" | "world":*)

...

2. "hello world foo" search
<img width="1274" alt="Screenshot 2024-11-22 at 7 57 10 AM" src="https://github.com/user-attachments/assets/21fd9657-58b5-4823-b103-8d8f9715174c">
PgTerm from log: ("hello" | "hello":*)&("world" | "world":*)&("foo" | "foo":*)
